contracts/contracts/vault/VaultCore.sol

```diff
@@ -390,6 +395,76 @@ contract VaultCore is VaultStorage {
         return vaultValue;
     }
 
+    /**
+     * @notice Strategist swaps collateral assets sitting in the vault.
+     * @param _fromAsset The token address of the asset being sold by the vault.
+     * @param _toAsset The token address of the asset being purchased by the vault.
+     * @param _fromAssetAmount The amount of assets being sold by the vault.
+     * @param _minToAssetAmount The minimum amount of assets to be purchased.
+     * @param _data implementation specific data. eg 1Inch swap data
+     * @return toAssetAmount The amount of toAssets that was received from the swap
+     */
+    function swapCollateral(
+        address _fromAsset,
+        address _toAsset,
+        uint256 _fromAssetAmount,
+        uint256 _minToAssetAmount,
+        bytes calldata _data
+    )
+        external
+        whenNotCapitalPaused
+        nonReentrant
+        onlyGovernorOrStrategist
+        returns (uint256 toAssetAmount)
+    {
+        // Check fromAsset and toAsset are valid
+        Asset memory fromAssetConfig = assets[address(_fromAsset)];
+        Asset memory toAssetConfig = assets[_toAsset];
+        require(fromAssetConfig.isSupported, "From asset is not supported");
+        require(toAssetConfig.isSupported, "To asset is not supported");
+
+        uint256 toAssetBalBefore = IERC20(_toAsset).balanceOf(address(this));
+
+        // Transfer from assets to the swapper contract
+        IERC20(_fromAsset).safeTransfer(swapper, _fromAssetAmount);
+
+        // Call to the Swapper contract to do the actual swap
+        // slither-disable-next-line unused-return
+        ISwapper(swapper).swap(
+            _fromAsset,
+            _toAsset,
+            _fromAssetAmount - 1,
+            _minToAssetAmount,
+            _data
+        );
+
+        // Compute the change in asset balance held by the Vault
+        toAssetAmount =
+            IERC20(_toAsset).balanceOf(address(this)) -
+            toAssetBalBefore;
+
+        // Check the to assets returned is above slippage amount specified by the strategist
+        require(
+            toAssetAmount >= _minToAssetAmount,
+            "Strategist slippage limit"
+        );
+
+        // Check the slippage against the Oracle in case the strategist made a mistake or has become malicious.
+        // to asset amount = from asset amount * from asset price / to asset price
+        uint256 minOracleToAssetAmount = (_fromAssetAmount *
+            (1e4 - fromAssetConfig.allowedSwapSlippageBps) *
+            IOracle(priceProvider).price(_fromAsset)) /
+            (IOracle(priceProvider).price(_toAsset) *
+                (1e4 + toAssetConfig.allowedSwapSlippageBps));
```

Comment:
I don't think we should consider the slippage twice - for each of the assets present in the exchange. Normally when I think of slippage when it comes to a transaction I think of the price movement in a pool. Which would mean that slippage pertains to a pair of assets rather than 1 single asset. 

1. Since here we configure it per asset I think we should either rename it to `allowedSwapSlippagePortionBps` or `allowedHalfSwapSlippageBps` to try to communicate that the configured slippage represents only a portion of the total slippage in transaction. (though I like none of the name suggestions 🙂  )

2. Or leave it as is and change the calculation to: 
```
uint256 minOracleToAssetAmount = (_fromAssetAmount *
            IOracle(priceProvider).price(_fromAsset)) /
            (IOracle(priceProvider).price(_toAsset)) * 
            (1e4 - fromAssetConfig.allowedSwapSlippageBps);
```

3. If you want to account for different assets having different liquidity and the swap to account for both slippages when considering min amount
```
uint256 minOracleToAssetAmount = (_fromAssetAmount *
            (1e4 - fromAssetConfig.allowedSwapSlippageBps / 2) *
            IOracle(priceProvider).price(_fromAsset)) /
            (IOracle(priceProvider).price(_toAsset) *
                (1e4 + toAssetConfig.allowedSwapSlippageBps / 2));
```

I would go with option 3

Reply:
@sparrowDom I think we need different slippages for different assets. WETH swaps should be very close to the Oracle price while stETH and rETH may not.

I agree the name is misleading. How about `allowedOracleSlippageBps`? This makes it clearer that it it slippage from the Oracle price and not slippage on an AMM pool from the swap size.

Reply:
`allowedOracleSlippage` is much better. Maybe also `allowedAssetSlippage` isn't bad since it similarly communicates that it pertains to 1 asset rather than a pool (pair of assets)

Reply:
The field is on the `Asset` structure so I think it's clear it relates to an asset. eg `assets[i].allowedOracleSlippageBps`.

I'll go ahead with `allowedOracleSlippageBps`